modules/ROOT/pages/anypoint-mq/3.x/anypoint-mq-listener.adoc

```diff
@@ -0,0 +1,347 @@
+
+By default, the circuit breaking feature is disabled.
+
+=== Circuit Breaker States
+
+The circuit breaker has three states: Closed, Open, and Half Open. The behavior of the app changes based on the current state. See the https://docs.microsoft.com/en-us/azure/architecture/patterns/circuit-breaker[Microsoft Circuit Breaker pattern] for more information.
+
+image::anypoint-mq/3.x/amq-3x-subscriber-states-view.png[Circuit Breaker States Diagram]
+
+* *Closed*
++
+The starting state where the subscriber retrieves messages normally from MQ based on its configuration, effectively working as if the circuit breaker is not present.
++
+* *Closed-Open Transition*
++
+When the number of failures occurs in succession during message processing, without successes, and reaches the `errorsThreshold` value, the circuit breaker trips and the circuit breaker transitions to an Open state.
++
+Messages that were already dispatched to the flow then finish processing, regardless of whether the result is success or failure.
++
+Messages kept locally that are in-flight for the broker but haven't been dispatched yet are not acknowledged and returned to the queue for redelivery to another consumer.
++
+* *Open*
++
+The subscriber doesn't attempt to retrieve messages, and skips the iterations silently until the `tripTimeout` is reached.
++
+* *Half Open*
++
+After the `tripTimeout` elapses, the subscriber goes to a Half Open state. In the next poll for messages, the subscriber retrieves a single message from the service and uses that message to check if the system has recovered before going back to the normal Closed state.
++
+When the subscriber successfully fetches a single message, dispatches it to the flow, and processing finishes successfully, the subscriber returns to normal and immediately attempts to fetch more messages.
++
+If Mule flow processing fails with one of the expected `onErrorTypes`, the circuit goes back to an Open state and resets the `tripTimeout` timer.
+
+=== Configure the Circuit Breaker
+
+You can configure a Circuit Breaker as either a Global Circuit Breaker or a Private Circuit Breaker.
+
+Either way, the configuration parameters are the same:
+
+* `onErrorTypes`
++
+The error types that count as a failure during the flow execution. An error occurrence counts only when the flow finishes with an error propagation. By default, all errors count as a circuit failure.
+* `errorsThreshold`
++
+The number of `onErrorTypes` errors that must occur for the circuit breaker to open.
+* `tripTimeout`
++
+How long the circuit remains open once the `errorsThreshold` is reached.
+* `circuitName`
++
+The name of a circuit breaker to bind to this configuration. By default, each queue has its own circuit breaker.
+
+
+==== Global Circuit Breaker
+
+Use a Global Circuit Breaker when you want to share the circuit state across multiple subscribers, as if subscribers are part of the same "circuit".
```

Comment:
Can we lowercase "global circuit breaker"?

period goes within quotes : "circuit."